lib/XTR/ScheduleReminderTask.php

```diff
@@ -0,0 +1,192 @@
+    private $userFactory;
+
+    /** @var ScheduleFactory */
+    private $scheduleFactory;
+
+    /** @var CampaignFactory */
+    private $campaignFactory;
+
+    /** @var ScheduleReminderFactory */
+    private $scheduleReminderFactory;
+
+    /** @var NotificationFactory */
+    private $notificationFactory;
+
+    /** @inheritdoc */
+    public function setFactories($container)
+    {
+        $this->date = $container->get('dateService');
+        $this->userFactory = $container->get('userFactory');
+        $this->scheduleFactory = $container->get('scheduleFactory');
+        $this->campaignFactory = $container->get('campaignFactory');
+        $this->scheduleReminderFactory = $container->get('scheduleReminderFactory');
+        $this->notificationFactory = $container->get('notificationFactory');
+        return $this;
+    }
+
+    /** @inheritdoc */
+    public function run()
+    {
+        $this->runMessage = '# ' . __('Schedule reminder') . PHP_EOL . PHP_EOL;
+
+        // Long running task
+        set_time_limit(0);
+
+        $this->runScheduleReminder();
+    }
+
+    /**
+     *
+     */
+    private function runScheduleReminder()
+    {
+        // Get all schedules
+        $schedules = $this->scheduleFactory->query();
+
+        foreach($schedules as $schedule) {
+
+            // Get all reminders of the schedule
+            $reminders = $this->scheduleReminderFactory->query(null, ['eventId' => $schedule->eventId]);
+
+            foreach ($reminders as $reminder) {
+
+                $now = $this->date->parse();
+                if ($reminder->reminderDt <= $now->format('U')) {
+
+                    $schedule->setCampaignFactory($this->campaignFactory);
+                    $title = $schedule->getEventTitle();
+
+                    switch ($reminder->type) {
+                        case ScheduleReminder::$TYPE_MINUTE:
+                            $type = ScheduleReminder::$MINUTE;
+                            $typeText = 'Minute(s)';
+                            break;
+                        case ScheduleReminder::$TYPE_HOUR:
+                            $type = ScheduleReminder::$HOUR;
+                            $typeText = 'Hour(s)';
+                            break;
+                        case ScheduleReminder::$TYPE_DAY:
+                            $type = ScheduleReminder::$DAY;
+                            $typeText = 'Day(s)';
+                            break;
+                        case ScheduleReminder::$TYPE_WEEK:
+                            $type = ScheduleReminder::$WEEK;
+                            $typeText = 'Week(s)';
+                            break;
+                        case ScheduleReminder::$TYPE_MONTH:
+                            $type = ScheduleReminder::$MONTH;
+                            $typeText = 'Month(s)';
+                            break;
+                        default:
+                            throw new \Xibo\Exception\NotFoundException('Unknown type');
```

Comment:
We might want to log and `continue` here, otherwise an incorrect reminder would stop all reminders from processing until it was fixed (and there isn't enough info here to try and fix)